watchtower-plugin/src/dbm.rs

```diff
@@ -207,12 +240,24 @@ impl DBM {
             let tower_id = TowerId::from_slice(&raw_towerid).unwrap();
             let net_addr: String = row.get(1).unwrap();
             let available_slots: u32 = row.get(2).unwrap();
-            let subscription_expiry: u32 = row.get(3).unwrap();
+
+            let mut stmt = self
+            .connection
+            .prepare("SELECT subscription_start, subscription_expiry FROM registration_receipts WHERE tower_id = ?1 AND subscription_expiry = (SELECT MAX(subscription_expiry) FROM registration_receipts WHERE tower_id = ?1)")
```

Comment:
Umm, this also passes tests:

```
SELECT tw.tower_id, tw.net_addr, tw.available_slots, rr.subscription_start, rr.max_se
FROM towers AS tw
JOIN (SELECT tower_id, subscription_start, MAX(subscription_expiry) AS max_se
      FROM registration_receipts
      GROUP BY tower_id) AS rr
ON (tw.tower_id = rr.tower_id)
```

Meaning the tests are insufficient, or the query can be simplified

Reply:
Do double check, b/c my SQL is a bit Rusty, but it is indeed passing the current tests for `watchtower-plugin`

Reply:
In [`here`](https://dba.stackexchange.com/questions/141594/how-select-column-does-not-list-in-group-by-clause) and [`here`](https://tableplus.com/blog/2019/10/select-column-not-in-group-by.html), this gives an error. So it's only SQLite being nice and fulfilling the query.
The tests are indeed not covering the case where more than one registration per tower exists.

Reply:
Aside from the test not being hard enough. Your query would actually work even on the hardened test.
I see this is because the grouping function used is `MAX()`. I tried doing a similar thing with `COUNT()` and it failed.
I wouldn't count on it though, cuz we might use another SQL engine later which would reject this query.

Reply:
> The tests are indeed not covering the case where more than one registration per tower exists.
In here I meant the test `test_store_load_towers`, which should test the mentioned case.

Reply:
After hardening the test, nope.
Because the thing about the available slots being overwritten with random numbers.
the [hardened test](https://github.com/meryacine/rust-teos/commit/25e0dbc04336b5dfa38ab8f0d1b3d0fc66bf1d70#diff-3af2742aaf4be3e870d3a4e8fab4d0b8990e271654af4e74b9eacb62db0df72cL696-L719) would fail with the 2 sql queries above and with the current code too.